crates/math/src/arith_expr.rs

```diff
@@ -116,60 +124,112 @@ impl<F: Field> ArithExpr<F> {
 	/// * [`Error::IncorrectArgumentLength`] if indices has length less than the current number of
 	///   variables
 	pub fn remap_vars(self, indices: &[usize]) -> Result<Self, Error> {
```

Comment:
This is complicated now. We should just move it to `ArithCircuit`, where it will be easier to implement.

That can be a follow-up.

Reply:
Within this change we need to have it in `ArithExpr` as it is still in use

Reply:
ArithExpr::remap_vars can call `ArithCircuit::new` -> `remap_vars` -> `ArithCircuit::into` for backwards compatibility.